diff -ur libid3tag-0.16.1.orig/compat.c libid3tag-0.16.1/compat.c
--- libid3tag-0.16.1.orig/compat.c	2021-08-15 13:52:07.000000000 -0700
+++ libid3tag-0.16.1/compat.c	2022-07-15 04:27:32.000000000 -0700
@@ -439,6 +439,10 @@
     encoding = id3_parse_uint(&data, 1);
     string   = id3_parse_string(&data, end - data, encoding, 0);
 
+    if (string == 0) {
+      continue;
+    }
+
     if (id3_ucs4_length(string) < 4) {
       free(string);
       continue;
